sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/encoders/Encoder.scala

```diff
@@ -31,6 +31,7 @@ import org.apache.spark.sql.types.StructType
  * and reuse internal buffers to improve performance.
  */
 trait Encoder[T] {
```

Comment:
Everything in `catalyst.*` is considered unstable and is excluded from the generated documentation.  So, usually we don't mark them developer API too.

However, since this is a user facing concept, we'll need to move at least `trait Encoder[T]` before the release.  I don't know where we want to put it though, `org.apache.spark.sql`? `org.apache.spark`?